### PR TITLE
Upgrade to SBT 1.0 and add Scala 2.12 support

### PR DESCRIPTION
**Pull request by jypma**
Also sanatizes the build structure quite a bit.

### CODE REVIEW

**Comment by jypma** [comment]
CC @luis-almeida-santos I believe you were asking for this ;-) We're looking into akka 2.5 as well.

File: build.sbt [hunk 1]
```diff
@@ -1,42 +1,52 @@
-scalaVersion := "2.11.8"
 
-import sbtprotobuf.{ProtobufPlugin=>PB}
+val akkaVersion = "2.4.18"
+val akkaHttpVersion = "10.0.7"
+val akkaInMemory = "com.github.dnvriend" %% "akka-persistence-inmemory" % "2.4.18.1"
+val kamonVersion = "0.6.6"
+val assertJ = "org.assertj" % "assertj-core" % "3.2.0"
```
Comment (ayurchuk) [inline build.sbt]: Can we keep only versions here?

File: build.sbt [hunk 2]
```diff
@@ -46,55 +56,170 @@ lazy val commonSettings = projectSettings ++ Seq(
...
+  .dependsOn(`ts-reaktive-marshal-akka`, `ts-reaktive-testkit` % "test", `ts-reaktive-testkit-assertj` % "test")
+
+lazy val `ts-reaktive-cassandra` = project
+  .settings(commonSettings: _*)
+  .settings(javaSettings: _*)
+  .settings(
+    libraryDependencies ++= Seq(
+      "com.readytalk" % "metrics3-statsd" % "4.1.0" // to log cassandra (codahale / dropwizard) metrics into statsd
+      //TODO check    "com.fasterxml.jackson.core" % "jackson-databind" % "2.7.4",
+      //TODO check  "com.google.protobuf" % "protobuf-java" % "2.6.1",
```
Comment (ayurchuk) [inline build.sbt]: When should these TODOs be considered?

File: ts-reaktive-backup/src/main/java/com/tradeshift/reaktive/backup/S3.java [hunk 3]
```diff
@@ -182,156 +134,28 @@ public static Instant getStartInstant(S3Entry entry) {
...
-                    .thenApply(response -> {
-                        if (response.nextContinuationToken.isDefined()) {
-                            return Optional.of(Pair.create(response.nextContinuationToken.get(), response.entries));
-                        } else {
-                            return Optional.of(Pair.create("done", response.entries));
-                        }
-                    });
-            }
-        }).flatMapConcat(seq -> Source.from(seq));
+    public Source<ListBucketResultContents, NotUsed> list(String keyPrefix) {
```
Comment (ayurchuk) [inline ts-reaktive-backup/src/main/java/com/tradeshift/reaktive/backup/S3.java]: `keyPrefix` is not used

File: build.sbt [hunk 4]
```diff
@@ -46,55 +56,170 @@ lazy val commonSettings = projectSettings ++ Seq(
...
+  )
+  .dependsOn(`ts-reaktive-marshal-akka`, `ts-reaktive-testkit` % "test", `ts-reaktive-testkit-assertj` % "test")
+
+lazy val `ts-reaktive-cassandra` = project
+  .settings(commonSettings: _*)
+  .settings(javaSettings: _*)
+  .settings(
+    libraryDependencies ++= Seq(
+      "com.readytalk" % "metrics3-statsd" % "4.1.0" // to log cassandra (codahale / dropwizard) metrics into statsd
+      //TODO check    "com.fasterxml.jackson.core" % "jackson-databind" % "2.7.4",
```
Comment (alar17) [inline build.sbt]: Remove these TODOs

**Comment by jypma** [comment]
This should cover https://github.com/Tradeshift/ts-reaktive/pull/70 and https://github.com/Tradeshift/ts-reaktive/pull/72 as well. @pjfanning

File: build.sbt (hunk 1 above)
Comment (jypma) [reply build.sbt]: why? then you'd have to repeat the artifact group and name in all places.

File: build.sbt (hunk 2 above)
Comment (jypma) [reply build.sbt]: they should be removed.

File: build.sbt [hunk 5]
```diff
@@ -46,55 +56,170 @@ lazy val commonSettings = projectSettings ++ Seq(
...
       "com.typesafe.akka" %% "akka-persistence-cassandra" % "0.29",
-      "com.readytalk" % "metrics3-statsd" % "4.1.0", // to log cassandra (codahale / dropwizard) metrics into statsd
+      "org.slf4j" % "slf4j-log4j12" % "1.7.12"
+    )
+  }  
+)
+
+lazy val kamonSettings = Seq(
+  libraryDependencies ++= {
+    val kamonVersion = "0.6.6"
```
Comment (ayurchuk) [inline build.sbt]: Can we drop duplicate `kamonVersion`?

File: ts-reaktive-backup/src/main/java/com/tradeshift/reaktive/backup/S3.java (hunk 3 above)
Comment (jypma) [reply ts-reaktive-backup/src/main/java/com/tradeshift/reaktive/backup/S3.java]: whoopsie.

File: build.sbt (hunk 5 above)
Comment (jypma) [reply build.sbt]: we probably should :)

File: build.sbt (hunk 1 above)
Comment (ayurchuk) [reply build.sbt]: It makes sense to keep them here. There is another one `slf4j-log4j12`.

**Comment by ayurchuk** [comment]
:star:

**Pull request merged**